### PR TITLE
Fire ball

### PR DESCRIPTION
**Pull request by pradzaa2k**

### CODE REVIEW

File: Assets/Scripts/DemonMovement.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public Transform startPoint;
+    public Transform endPoint;
+    public float speed = 5.0f;
+    private int destination  = 0;
+    // Update is called once per frame
+    void Update()
+    {
+        if (destination == 0)
+        {
+            transform.position = Vector2.MoveTowards(transform.position, startPoint.position, speed * Time.deltaTime);
```
Comment (Nivedit24) [inline Assets/Scripts/DemonMovement.cs:16]: Use the floor prefab and apply the TornadoMovement script to Demon. TornadoMovement can be renamed to EnemyMovement and used for both tornado and demons. When enemy collides with edge in floor prefab direction gets reversed. This script can be deleted

File: Assets/Scripts/FireProjectile.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            Instantiate(fireballPrefab, launchPointRight.position, launchPointRight.rotation);
+            shootTime = 1;
+            remainingFireballs -= 1;
+        }
+        if (Input.GetKeyDown(KeyCode.LeftShift) && shootTime <= 0 && !playerMovement.faceRight)
+        {
+            Instantiate(fireballPrefab, launchPointLeft.position, launchPointLeft.rotation);
+            shootTime = 1;
+            remainingFireballs -= 1;
+        }
```
Comment (Nivedit24) [inline Assets/Scripts/FireProjectile.cs:36]: Can be simplified
`if (Input.GetKeyDown(KeyCode.LeftShift) && shootTime <= 0)
        {
            if(playerMovement.faceRight)
                   Instantiate(fireballPrefab, launchPointRight.position, launchPointRight.rotation);
            else
                  Instantiate(fireballPrefab, launchPointLeft.position, launchPointLeft.rotation);
            shootTime = 1;
            remainingFireballs -= 1;
        }`

File: Assets/Scripts/FireProjectile.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        shootTime -= Time.deltaTime;
```
Comment (Nivedit24) [inline Assets/Scripts/FireProjectile.cs:37]: Whats this used for? Preferably I want to shoot without much delay

File: Assets/Scripts/VolcanoProjectile.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    void setVelocity()
+    {
+        volcanoBall.velocity = transform.right * speed;
```
Comment (Nivedit24) [inline Assets/Scripts/VolcanoProjectile.cs:24]: Can we have random speeds within a defined range here? Will be more fun

**Review by Nivedit24 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/DemonMovement.cs (hunk 1 above)
Comment (pradzaa2k) [reply Assets/Scripts/DemonMovement.cs]: Implemented

File: Assets/Scripts/PlayerMovement.cs [hunk 5]
```diff
@@ -90,6 +88,14 @@ void Update()
             }
         }
 
+        if (direction < 0)
+        {
+            faceRight = false;
+        }
+        else if (direction > 0)
+        {
+            faceRight = true;
```
Comment (Nivedit24) [inline Assets/Scripts/PlayerMovement.cs:97]: Can be simplified to faceright = direction >= 0

File: Assets/Scripts/PlayerMovement.cs [hunk 6]
```diff
@@ -194,6 +200,31 @@ void OnCollisionEnter2D(Collision2D collision)
...
+                    collision.gameObject.SetActive(false);
+                }
+                if (instructions.Contains(collision.gameObject))
+                {
+                    DisplayText("Press Shift to shoot in player's direction", collision.gameObject);
+                }
+                break;
+            case "VolcanoBall":
+                Debug.Log("Hit by volcanoBall");
+                playerReceiver.TakeDamage(50);
```
Comment (Nivedit24) [inline Assets/Scripts/PlayerMovement.cs:226]: Can we differentiate Fireballs from Demon and Lava from Volcano. Currently they look same but damage is different

File: Assets/Scripts/Projectile.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        if (playerMovement != null)
+        {
+            if (playerMovement.faceRight)
+            {
+                fireBall.velocity = new Vector2(speed, fireBall.velocity.y);
+            }
+            else
+            {
+                fireBall.velocity = new Vector2(-speed, fireBall.velocity.y);
```
Comment (Nivedit24) [inline Assets/Scripts/Projectile.cs:45]: Can be simplified as above

File: Assets/Scripts/SwitchMovement.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            if (!activated)
+            {
+                activated = true;
+                switches.GetComponent<Renderer>().material.color = Color.green;
+            }
+            else
+            {
+                activated = false;
+                switches.GetComponent<Renderer>().material.color = Color.red;
```
Comment (Nivedit24) [inline Assets/Scripts/SwitchMovement.cs:49]: activated = !activated;
switches.GetComponent<Renderer>().material.color = activated ? Color.green : Color.Red;

**Review by Nivedit24 — CHANGES_REQUESTED** [review]

**Review by Nivedit24 — APPROVED** [review]
LGTM

**Pull request merged**